### PR TITLE
[IOTDB-5284] Fix some sonar bugs

### PR DESCRIPTION
**Pull request by BUAAserein**
##### Key changed in this PR
- Math operands should be cast before assignment
- Resources should be closed

### CODE REVIEW

File: example/session/src/main/java/org/apache/iotdb/SessionExample.java [hunk 1]
```diff
@@ -828,18 +828,19 @@ private static void nonQuery() throws IoTDBConnectionException, StatementExecuti
...
-  private static void setTimeout() throws StatementExecutionException {
-    Session tempSession = new Session(LOCAL_HOST, 6667, "root", "root", 10000, 20000);
-    tempSession.setQueryTimeout(60000);
+  private static void setTimeout() throws StatementExecutionException, IoTDBConnectionException {
+    try (Session tempSession = new Session(LOCAL_HOST, 6667, "root", "root", 10000, 20000)) {
+      tempSession.setQueryTimeout(60000);
+    }
   }
 
   private static void createClusterSession() throws IoTDBConnectionException {
```
Comment (OneSizeFitsQuorum) [inline example/session/src/main/java/org/apache/iotdb/SessionExample.java:837]: We can just remove this function now as it's used for old cluster module

File: node-commons/src/main/java/org/apache/iotdb/commons/udf/builtin/UDTFEqualSizeBucketOutlierSample.java [hunk 2]
```diff
@@ -273,7 +273,7 @@ public void outlierSampleDouble(RowWindow rowWindow, PointCollector collector)
           new PriorityQueue<>(number, Comparator.comparing(o -> o.right));
 
       double A = row0y - row1y;
```
Comment (OneSizeFitsQuorum) [inline node-commons/src/main/java/org/apache/iotdb/commons/udf/builtin/UDTFEqualSizeBucketOutlierSample.java:275]: why miss this line?

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileRestorableReader.java [hunk 3]
```diff
@@ -61,11 +61,12 @@ private void checkAndRepair() throws IOException {
...
-      RestorableTsFileIOWriter rWriter =
-          new RestorableTsFileIOWriter(FSFactoryProducer.getFSFactory().getFile(file));
-      TsFileWriter writer = new TsFileWriter(rWriter);
-      // This writes the right magic string
-      writer.close();
+      try (RestorableTsFileIOWriter rWriter =
+          new RestorableTsFileIOWriter(FSFactoryProducer.getFSFactory().getFile(file))) {
+        TsFileWriter writer = new TsFileWriter(rWriter);
+        // This writes the right magic string
+        writer.close();
```
Comment (OneSizeFitsQuorum) [inline tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileRestorableReader.java:68]: remove this line?

File: node-commons/src/main/java/org/apache/iotdb/commons/udf/builtin/UDTFEqualSizeBucketOutlierSample.java (hunk 2 above)
Comment (BUAAserein) [reply node-commons/src/main/java/org/apache/iotdb/commons/udf/builtin/UDTFEqualSizeBucketOutlierSample.java:275]: Please see line 269 and 270. They are already of type double.

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileRestorableReader.java (hunk 3 above)
Comment (BUAAserein) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileRestorableReader.java:68]: writer will not close automatically

File: example/session/src/main/java/org/apache/iotdb/SessionExample.java (hunk 1 above)
Comment (BUAAserein) [reply example/session/src/main/java/org/apache/iotdb/SessionExample.java:837]: ok

Comment (OneSizeFitsQuorum) [reply example/session/src/main/java/org/apache/iotdb/SessionExample.java:831]: do not remove `setTimeout`, remove `createClusterSession`

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileRestorableReader.java (hunk 3 above)
Comment (OneSizeFitsQuorum) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileRestorableReader.java:68]: Who not remove `TsFileWriter writer = new TsFileWriter(rWriter);` into `try-with-resource` ?

File: example/session/src/main/java/org/apache/iotdb/SessionExample.java (hunk 1 above)
Comment (BUAAserein) [reply example/session/src/main/java/org/apache/iotdb/SessionExample.java]: ok

File: tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileRestorableReader.java (hunk 3 above)
Comment (BUAAserein) [reply tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileRestorableReader.java]: ok

**Review by OneSizeFitsQuorum — APPROVED** [review]
LGTM

**Review by CRZbulabula — APPROVED** [review]
LGTM～

**Review by SzyWilliam — APPROVED** [review]
LGTM

**Pull request merged**